test/cpu_any.hpp

```diff
@@ -0,0 +1,82 @@
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#pragma once
+
+#include "tensor_view.hpp"
+#include <ford.hpp>
+#include <tensor_holder.hpp>
+#include <miopen/tensor_view_utils.hpp>
+
+template <class T>
+void cpu_any_forward(tensor<T> input, tensor<unsigned char>& ref_output, size_t dim, bool keepdim)
+{
+    auto input_tv  = miopen::get_inner_expanded_tv<5>(input.desc);
+    auto output_tv = miopen::get_inner_expanded_tv<5>(ref_output.desc);
+
+    auto input_dims  = input.desc.GetLengths();
+    auto output_dims = ref_output.desc.GetLengths();
+
+    auto output_numel = ref_output.desc.GetElementSize();
+    auto input_numel  = input.desc.GetElementSize();
+
+    auto reduce_size = input_dims[dim];
+
+    auto inner_size = 1ULL;
+    for(int32_t i = dim + 1; i < input_dims.size(); i++)
+    {
+        inner_size *= input_dims[i];
+    }
+
+    if(dim != -1)
+    {
+        par_ford(output_numel)([&](size_t o) {
+            size_t idx        = (o / inner_size) * inner_size * reduce_size + o % inner_size;
+            auto inp_tl       = tensor_layout_t(input_tv, idx);
+            auto input_idx    = input_tv.get_tensor_view_idx(inp_tl);
+            unsigned char any = 0;
+            ford(reduce_size)([&](size_t o) {
+                unsigned char val = input[input_idx] != 0;
+                any               = any || val;
+                input_idx += inner_size;
+            });
+
+            auto out_tl         = tensor_layout_t(output_tv, o);
+            auto out_idx        = output_tv.get_tensor_view_idx(out_tl);
+            ref_output[out_idx] = any;
+        });
+    }
+    else
+    {
+        unsigned char any = 0;
+        par_ford(input_numel)([&](size_t i) {
+            auto inp_tl       = tensor_layout_t(input_tv, i);
+            auto input_idx    = input_tv.get_tensor_view_idx(inp_tl);
+            unsigned char val = input[input_idx] != 0;
+            any               = any || val;
+        });
+        ref_output[0] = any;
+    }
```

Comment:
any was init outside the loop and update inside each one, please check for correction

Reply:
This is just the same as calculating sum. `any` is initialized outside the loop and accumulates the result of the logical OR operation over all elements in the `reduce_size` dimension. If any non-zero value is found, any will be set to 1 (true), and this value will be assigned to the corresponding element in the `ref_output` array.